internal/models/valueset.xo.go

```diff
@@ -132,36 +134,22 @@ func (vs *ValueSet) Delete(ctx context.Context, db DB) error {
 	vs._deleted = true
 	return nil
 }
-
-// ValueSetByID retrieves a row from 'public.value_set' as a [ValueSet].
-//
-// Generated from index 'value_set_id_key'.
-func ValueSetByID(ctx context.Context, db DB, id string) (*ValueSet, error) {
-	// query
-	const sqlstr = `SELECT ` +
-		`oid, id, name, code, status, definitiontext, scopenotetext, assigningauthorityid, legacyflag, statusdate ` +
-		`FROM public.value_set ` +
-		`WHERE id = $1`
-	// run
-	logf(sqlstr, id)
-	vs := ValueSet{
-		_exists: true,
-	}
-	if err := db.QueryRowContext(ctx, sqlstr, id).Scan(&vs.Oid, &vs.ID, &vs.Name, &vs.Code, &vs.Status, &vs.Definitiontext, &vs.Scopenotetext, &vs.Assigningauthorityid, &vs.Legacyflag, &vs.Statusdate); err != nil {
-		return nil, logerror(err)
-	}
-	return &vs, nil
-}
-
 // ValueSetByOid retrieves a row from 'public.value_set' as a [ValueSet].
 //
 // Generated from index 'value_set_pkey'.
+//
+// ValueSetByOid
 func ValueSetByOid(ctx context.Context, db DB, oid string) (*ValueSet, error) {
+	fmt.Println("kcd", oid)
+	dbId, err := DetermineIdType(oid)
+	if err != nil {
+		return nil, logerror(err)
+	}
 	// query
-	const sqlstr = `SELECT ` +
+	var sqlstr = `SELECT ` + 
 		`oid, id, name, code, status, definitiontext, scopenotetext, assigningauthorityid, legacyflag, statusdate ` +
 		`FROM public.value_set ` +
-		`WHERE oid = $1`
+		`WHERE ` + dbId + ` = $1`
```

Comment:
Instead of building a string like this, could you add another placeholder `$2` for the oid/id column?

Reply:
and then we can scrap the `DetermineIdType` method

Reply:
Ahhh yes, that would work as well! Always better to do it in raw SQL if possible, good thought!

Reply:
is removal of the `DetermineIdType` method within scope for this PR?

Reply:
@akasper it was originally added in this PR, so it's like it never happened 🪄